Add improvements to modules, services, and contributions pages (#889)
gruntwork-io/docs: _docs-sources/library/overview/services.md

# What is a service module?

> **Service modules are opinionated combinations of ["building block" modules](./modules). They are usually written in Terraform, represent a full use case, and are designed to be used "off the shelf."**

Reviewer: I see what you're trying to do with shortening the description, but I don't think it's quite getting the point across. What about this?

```suggestion
Service modules are combinations of [modules](./modules) covering a wide variety of use cases, including CI/CD, networking, container orchestration (EKS, ECS), and more. Service modules are usually written in Terraform, represent a full use case, and are designed to be used "off the shelf."
```

Regarding style — I don't think we need to put this in bold, it's the first sentence of the first section on the page. We intentionally removed bold text in many other places on the docs site.

## Example

> Let’s look at an example service module. The [rds service module](/reference/services/data-storage/amazon-rds) creates not only an RDS database, but also a DNS record (so you can access the database with a human-friendly name), CloudWatch alarms (to alert you if something goes wrong), a CloudWatch dashboard (so you can see RDS KPIs at a glance), and a strategy for disaster recovery. The module does this by using many "building block" modules.

Reviewer: ```suggestion
Let’s look at an example service module. The [rds service module](/reference/services/data-storage/amazon-rds) creates an RDS database, a DNS record in Amazon Route53, CloudWatch alarms, a CloudWatch dashboard, and scheduled snapshots of the database. 

After applying, you will have an RDS instance that you can access via a human-friendly name, alarms to alert you if something goes wrong, a dashboard for viewing RDS KPIs, and a strategy for disaster recovery.
```

All of the parenthesis made this hard to read, so perhaps splitting this into two small paragraphs is more powerful?

Also, I don't think we need to redefine what a service module is, it's at the top of the page.

Author: Fixed in 54c7c0a705802bd656e58b914ad000e08b22640c!

## Service modules are optimized for convenience

> Whereas "building block" modules are optimized for control, service modules are optimized for convenience. That is, a service module reflects a complete Gruntwork opinion on how you should run a piece of infrastructure. If you agree with our opinion, using a service module gives you high convenience because you have a single Terraform module that does everything you need and which Gruntwork will maintain!

Reviewer: ```suggestion
Whereas "building block" modules are optimized for control, service modules are optimized for convenience.

A service module reflects a complete Gruntwork opinion on how you should run a piece of infrastructure. Using a service module gives you the convenience of a single Terraform module that does everything you need, with Gruntwork maintaining the code.
```

## When to use a service module

Using a service module can save you time because Gruntwork has already pieced together individual building block modules and tested that they correctly reference each other.

For example, the [eks-cluster service module](/reference/services/app-orchestration/amazon-eks) combines all the modules you need to run an EKS (Kubernetes) cluster in a typical production environment, including modules for the control plane, worker nodes, secrets management, log aggregation, alerting, and so on.

If you need more flexibility than our services provide, then you can combine modules from our [module catalog](./modules#where-to-find-the-building-block-modules), your own modules, or open source modules to meet your specific use case.

CIS customers also have access to the [terraform-aws-cis-service-catalog git repo](https://github.com/gruntwork-io/terraform-aws-service-catalog) to help ensure conformity to the [CIS AWS Foundations Benchmark](https://gruntwork.io/achieve-compliance/).

## Where to find the service modules

The service catalog features over 25 service modules spanning three major use cases:

1. AWS foundations

> 2. Run my app

Reviewer: ```suggestion
2. Running applications
```

> 3. Store my data

Reviewer: ```suggestion
3. Storing data
```

       if there is no example?](https://github.com/gruntwork-io/knowledge-base/discussions/360#discussioncomment-25705480).

1. `test`: Automated tests for the code in modules and examples.

## How service modules are updated

> When [building block modules are updated](/library/overview/modules#how-modules-are-updated), we propagate those updates to the service catalog. Keep in mind that there is sometimes a delay between when a "building block" module is updated, and when that update is reflected in the service catalog.

Reviewer: ```suggestion
When [building block modules are updated](/library/overview/modules#how-modules-are-updated), we propagate those updates to the service catalog. Keep in mind that there is sometimes a delay between when a "building block" module is updated and when that update is reflected in the service catalog.
```
